usb_protocol/emitters/descriptors/hid.py

```diff
@@ -0,0 +1,109 @@
+import unittest
+
+from contextlib import contextmanager
+
+from ..           import emitter_for_format
+from ..descriptor import ComplexDescriptorEmitter
+from ...types.descriptors.hid import \
+    HIDDescriptor as HIDDescriptorType
+from ...types.descriptors.hid import *
+
+ReportDescriptorEmitter = emitter_for_format(ReportDescriptor)
+
+_hid_item_length = [ 0, 1, 2, 4 ]
+
+class HIDDescriptor(ComplexDescriptorEmitter):
+    DESCRIPTOR_FORMAT = HIDDescriptorType
+
+    def add_report_raw(self, report_data):
+        """Append raw report item or bytes to HID report
+
+        Arguments:
+        report_data -- bytes-like or ReportDescriptor to be appended to
+                       the HID report.
+        """
+        self._reports.append(hid_data)
```

Comment:
This seems like a typo -- seems like you want `.append(report_data)`, not `hid_data`.